### PR TITLE
More structured logger names, #27922

### PR DESCRIPTION
**Pull request by patriknw**
Starting with Akka Cluster to see if we agree with the approach.

Refs #27922

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterLogClass.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package akka.cluster
+
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object ClusterLogClass {
```
[the remark this answers is not in the document]
Comment (chbatey) [reply akka-cluster/src/main/scala/akka/cluster/ClusterLogClass.scala]: :+1:

**Review by chbatey — APPROVED** [review]
Looks like a good approach 

Will do the same in akka management for bootstrap

File: akka-actor/src/main/scala/akka/event/Logging.scala [hunk 2]
```diff
@@ -270,6 +270,11 @@ trait LoggingBus extends ActorEventBus {
   def getClazz(t: T): Class[_] = t.getClass
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] final case class ActorWithLogClass(actor: Actor, logClass: Class[_])
```
[the remark this answers is not in the document]
Comment (chbatey) [reply akka-actor/src/main/scala/akka/event/Logging.scala]: aha so we get path and class, nice

[the remark this answers is not in the document]
Comment (chbatey) [reply akka-actor/src/main/scala/akka/event/Logging.scala]: Could this be public? Or internal stable API so we can use it other akka projects?

[the remark this answers is not in the document]
Comment (patriknw) [reply akka-actor/src/main/scala/akka/event/Logging.scala]: @chbatey do you think we need to have this public, e.g. for usage from Bootstrap. Maybe the path is not important there?

[the remark this answers is not in the document]
Comment (patriknw) [reply akka-actor/src/main/scala/akka/event/Logging.scala]: > so we get path and class, nice

yes, we get a class from `ActorLogging` too, but that's always the actor's class and not customizable

[the remark this answers is not in the document]
Comment (patriknw) [reply akka-actor/src/main/scala/akka/event/Logging.scala]: ok, I'll make it public

[the remark this answers is not in the document]
Comment (chbatey) [reply akka-actor/src/main/scala/akka/event/Logging.scala]: > @chbatey do you think we need to have this public, e.g. for usage from Bootstrap. Maybe the path is not important there?

Actually yes it is less important there, more important for things like sharding where there can be multiple of them with different paths

**Comment by patriknw** [comment]
@chbatey I have pushed additional improvements, but realized that most logger names are actually good, when using ActorLogging since config can be setup on the package (prefix)

wdyt?

**Review by johanandren — APPROVED** [review]
LGTM

**Review by chbatey — COMMENTED** [review]
Maybe we only need the improvements to which logger is used in this PR given the package names work out?

**Comment by patriknw** [comment]
> Maybe we only need the improvements to which logger is used in this PR given the package names work out?

You mean that I should revert the cluster changes? I still think they are an improvement from before.

**Pull request merged**